### DISCUSSION TITLE
Rabbitmq installation with default configuration is not working in FIPS enabled OCP cluster

### DISCUSSION THREAD

**Discussion by manoj4rabbbitmq** [rabbitmq/rabbitmq-server#10585] [Other] [answered]
### Describe the bug

When the peer discovery mechanism used is rabbit_peer_discovery_k8s  . The installation was successful. But  when the rabbitmq pods are keep on restarting with the error  

*2022-11-14 07:42:49.501027+00:00 [error] <0.230.0> Failed to lock with peer discovery backend rabbit_peer_discovery_k8s: "{failed_connect,\n    [{to_address,{\"kubernetes.default.svc.cluster.local\",443}},\n     {inet,\n         [inet],\n         {eoptions,\n             {{badarg,\n                  [

{crypto,evp_generate_key_nif,[x25519,undefined],[]}
,\n                   {ssl_cipher,generate_client_shares,2,\n                       [

{file,\"ssl_cipher.erl\"}
,{line,1265}]},\n                   {ssl_gen_statem,initial_hello,3,\n                       [

{file,\"ssl_gen_statem.erl\"}
,{line,457}]},\n                   {gen_statem,loop_state_callback,11,\n                       [

{file,\"gen_statem.erl\"}
,{line,1203}]},\n                   {tls_connection,init,1,\n                       [

{file,\"tls_connection.erl\"}
,{line,154}]},\n                   {proc_lib,init_p_do_apply,3,\n                       [

{file,\"proc_lib.erl\"}
,{line,226}]}]},\n              {gen_statem,call,[<0.754.0>,

{start,2250}
,infinity]}}}}]}"* 

 
And we don't see this issue when we provide the peer discovery mechanism as classic config. We need to know why we are seeing this error, if the peer discovery mechanism is not classic config. Also what is the recommended peer discovery mechanism that should be used for rabbitmq?

There are reported problems in RabbitMQ open source to work with a FIPS enabled K8s cluster, when using the peer discovery k8s cluster formation method. 

The underlying problem is from Erlang, but there is no formal word about how to make this combination work. 

When can we expect Erlang will gain FIPS mode compatibility ?

### Reproduction steps

Install rabbitmq container in FIPS enabled OCP cluster by setting peer discovery mechanism is set to rabbit_peer_discovery_k8s

### Expected behavior

Successful installaton of rabbitmq.

**Comment by mkuratczyk** [comment, marked as answer]
You've already started a discussion and received an answer:
https://github.com/rabbitmq/rabbitmq-server/discussions/10575

**Comment by michaelklishin** [comment]
@manoj4rabbbitmq please stop repeatedly filing identical issues (that we then move to Discussions) or you will be banned from ever filing a new issue in this org.
